app/Models/ProviderUserMapping.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProviderUserMapping extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'user_id',
+        'provider_platform_id',
+        'external_user_id',
+        'external_username', // if the user has a field in the external platform that is different from 'username'
+        'authentication_provider_id',
```

Comment:
What is the field supposed to be and why do we need it? In the factory its a number, in the migration it is a string, and in a test it is "openid_connect".

Reply:
This is supposed to be the type of auth we are using with the provider, currently it defaults to openid_connect, but in the future it might be SAML or OAuth2. I'm going to make a new Enum for this, and we can enforce it like that, if that makes sense to you

Reply:
Just making a note that we talked about making this field 'authentication_provider_status'.